packages/react-devtools-shared/src/backend/DevToolsComponentStackFrame.js

```diff
@@ -85,14 +85,16 @@ export function describeNativeComponentFrame(
   Error.prepareStackTrace = undefined;
 
   reentry = true;
-  let previousDispatcher;
-  if (__DEV__) {
-    previousDispatcher = currentDispatcherRef.current;
-    // Set the dispatcher in DEV because this might be call in the render function
-    // for warnings.
-    currentDispatcherRef.current = null;
-    disableLogs();
-  }
+
+  // Override the dispatcher so effects scheduled by this shallow render are thrown away.
+  //
+  // Note that unlike the code this was forked from (in ReactComponentStackFrame)
+  // DevTools should override the dispatcher in production mode as well,
+  // because app code might log errors or warnings.
```

Comment:
I think this comment isn't entirely clear. I read "in production mode" as "when app runs in production mode". Maybe you could say "should override the dispatcher when DevTools is compiled in production mode, because the app itself may be in development mode and log errors".